projects/kit/components/progress/progress-segmented/progress-segmented.component.ts

```diff
@@ -0,0 +1,44 @@
+import {ChangeDetectionStrategy, Component, HostBinding, Input} from '@angular/core';
+import {tuiDefaultProp} from '@taiga-ui/cdk';
+import {TuiSizeS} from '@taiga-ui/core';
+
+export const nonNegativeInt = (value: number | string) =>
+    Number.isInteger(+value) && value >= 0;
```

Comment:
`+value` we do not convert types, if somebody passes a string let them see the assertion

Reply:
fixed